src/pool/poolClient.ts

```diff
@@ -217,9 +219,56 @@ export class PoolEventState {
 }
 
 class UserState {
-  constructor(private contract: uma.clients.erc20.Instance) {}
-  public async read(user: string) {
+  private seen = new Set<string>();
+  private events: uma.clients.erc20.Transfer[] = [];
+  constructor(private contract: uma.clients.erc20.Instance, private startBlock = 0) {}
+  private makeId(params: EventIdParams) {
+    return [params.blockNumber, params.transactionIndex, params.logIndex].join("!");
+  }
+  hasEvent(params: EventIdParams) {
+    return this.seen.has(this.makeId(params));
+  }
+  private addEvent(params: EventIdParams) {
+    return this.seen.add(this.makeId(params));
+  }
+  private filterSeen = (params: EventIdParams) => {
+    const seen = this.hasEvent(params);
+    if (!seen) this.addEvent(params);
+    return !seen;
+  };
+  public async readEvents(user:string, endBlock: number) {
+    if (endBlock <= this.startBlock) return [];
+    const events = (
+      await Promise.all([
+        ...(await this.contract.queryFilter(this.contract.filters.Transfer(user,undefined), this.startBlock, endBlock)),
+        ...(await this.contract.queryFilter(this.contract.filters.Transfer(undefined,user), this.startBlock, endBlock)),
+      ])
```

Comment:
We could end up with duplicates where a user transfers to themselves, right? Will that be a problem? Should we do a post-filter for events that are from and to the same address?

Reply:
yea can do that!